cmd/kubectl-datadog/metrics/get/get.go

```diff
@@ -0,0 +1,187 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	getExample = `
+  # view all DatadogMetric in the current namespace
+  %[1]s get
+
+  # get the DatadogMetric named foo
+  %[1]s get foo
+`
+)
+
+// options provides information required by Datadog get command
+type options struct {
+	genericclioptions.IOStreams
+	common.Options
+	args              []string
+	datadogMetricName string
+}
+
+// metricData provides information about a datadogmetric's status
+type metricData struct {
+	Namespace  string
+	Name       string
+	Active     string
+	Valid      string
+	Value      string
+	References string
+	UpdateTime string
+}
+
+// newOptions provides an instance of getOptions with default values
+func newOptions(streams genericclioptions.IOStreams) *options {
+	o := &options{
+		IOStreams: streams,
+	}
+	o.SetConfigFlags()
+	return o
+}
+
+// New provides a cobra command wrapping options for "get" sub command
+func New(streams genericclioptions.IOStreams) *cobra.Command {
+	o := newOptions(streams)
+	cmd := &cobra.Command{
+		Use:          "get [DatadogMetric name] [flags]",
+		Short:        "Get DatadogMetric deployment(s)",
+		Example:      fmt.Sprintf(getExample, "kubectl datadog metrics"),
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.complete(c, args); err != nil {
+				return err
+			}
+			if err := o.validate(); err != nil {
+				return err
+			}
+			return o.run()
+		},
+	}
+
+	o.ConfigFlags.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+// complete sets all information required for processing the command
+func (o *options) complete(cmd *cobra.Command, args []string) error {
+	o.args = args
+	if len(args) > 0 {
+		o.datadogMetricName = args[0]
+	}
+	return o.Init(cmd)
+}
+
+// validate ensures that all required arguments and flag values are provided
+func (o *options) validate() error {
+	if len(o.args) > 1 {
+		return errors.New("either one or no arguments are allowed")
+	}
+	return nil
+}
+
+// run runs the get command
+func (o *options) run() error {
+	ddList := &v1alpha1.DatadogMetricList{}
+	if o.datadogMetricName == "" {
+		if err := o.Client.List(context.TODO(), ddList, &client.ListOptions{Namespace: o.UserNamespace}); err != nil {
+			return fmt.Errorf("unable to list DatadogMetric: %v", err)
+		}
+	} else {
+		dd := &v1alpha1.DatadogMetric{}
+		err := o.Client.Get(context.TODO(), client.ObjectKey{Namespace: o.UserNamespace, Name: o.datadogMetricName}, dd)
+		if err != nil && apierrors.IsNotFound(err) {
+			return fmt.Errorf("DatadogMetric %s/%s not found", o.UserNamespace, o.datadogMetricName)
+		} else if err != nil {
+			return fmt.Errorf("unable to get DatadogMetric: %v", err)
+		}
+		ddList.Items = append(ddList.Items, *dd)
+	}
+
+	table := newTable(o.Out)
+	for _, item := range ddList.Items {
+		metric := &metricData{
+			Namespace: item.Namespace,
+			Name:      item.Name,
+		}
+		if item.Status.Conditions != nil {
+			for i := range item.Status.Conditions {
```

Comment:
nits: `range` support the fact that `item.Status.Conditions`, so no need to have `if item.Status.Conditions != nil {` before

```suggestion
		for i := range item.Status.Conditions {
```